### PR TITLE
Feat/pokebattle flow

### PR DESCRIPTION
**Pull request by naftalima**
## Objective and Motivation
- Change the flow of the Pokebattle to make it more coherent with the new models

### CODE REVIEW

File: backend/battles/pokemon.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+from urllib.parse import urljoin
```
Comment (rsarai) [inline backend/battles/pokemon.py:1]: Nothing here is being used right? You can add this in a different PR.

File: backend/battles/services/battle.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+def battle_round(creator_pkn, opponent_pkn):
```
Comment (rsarai) [inline backend/battles/services/battle.py:1]: Nothing here is being used right? You can add this in a different PR.

File: backend/battles/forms.py [hunk 3]
```diff
@@ -1,9 +1,15 @@
 from django import forms
 
-from .models import Battle
+from .models import Battle  # , Team
 
 
 class TrainersRoundForm(forms.ModelForm):
```
[the remark this answers is not in the document]
Comment (rsarai) [reply backend/battles/forms.py:6]: or just `BattleForm`

**Review by rsarai — CHANGES_REQUESTED** [review]
@naftalima I made a lot of suggestions regarding names. One of the things that makes a system complex is a need for extra information in order to complete a task. In this case, one was supposed to know that a battle has trainers in order to understand that SelectTrainers is in reality creates battles. I advise you to avoid this type of cognitive load. More info on this here: https://vinta.atlassian.net/wiki/spaces/development/blog/2021/05/05/1686667278/Symptoms+of+complexity

File: backend/battles/forms.py [hunk 4]
```diff
@@ -1,9 +1,15 @@
 from django import forms
 
-from .models import Battle
+from .models import Battle  # , Team
```
[the remark this answers is not in the document]
Comment (naftalima) [reply backend/battles/forms.py:3]: ok!

File: backend/battles/views.py [hunk 5]
```diff
@@ -1,24 +1,64 @@
...
+
+    def get_queryset(self):
+        queryset_filtered = Battle.objects.filter(
+            creator__exact=self.request.user
+        ) | Battle.objects.filter(opponent__exact=self.request.user)
+
+        queryset = {
+            "on_going": queryset_filtered.filter(winner__isnull=True),
+            "settled": queryset_filtered.filter(winner__isnull=False),
+        }
```
[the remark this answers is not in the document]
Comment (naftalima) [reply backend/battles/views.py:56]: No. I did it myself. should I change? and if so, what is wrong?

File: backend/battles/pokemon.py (hunk 1 above)
Comment (naftalima) [reply backend/battles/pokemon.py:1]: This code is **old** - I was afraid to lose it again.

File: backend/battles/services/battle.py (hunk 2 above)
Comment (naftalima) [reply backend/battles/services/battle.py:1]: same here

File: backend/battles/pokemon.py (hunk 1 above)
Comment (naftalima) [reply backend/battles/pokemon.py:1]: but I won't do it again

File: backend/battles/forms.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (naftalima) [reply backend/battles/forms.py]: this was an old code, but I will not send it again in PR

File: backend/battles/views.py [hunk 6]
```diff
@@ -1,24 +1,64 @@
...
+class BattlesView(ListView):  # pylint: disable=too-many-ancestors
+    model = Battle
+    template_name = "battles/battles.html"
+    context_object_name = "battles"
+    # TODO paginate_by = 10
+
+    def get_queryset(self):
+        queryset_filtered = Battle.objects.filter(
+            creator__exact=self.request.user
+        ) | Battle.objects.filter(opponent__exact=self.request.user)
```
[the remark this answers is not in the document]
Comment (naftalima) [reply backend/battles/views.py:51]: Cool! Good to know. I will use this Q object.

[the remark this answers is not in the document]
Comment (naftalima) [reply backend/battles/views.py]: done

**Review by rsarai — CHANGES_REQUESTED** [review]

File: backend/battles/views.py [hunk 7]
```diff
@@ -1,24 +1,65 @@
...
-class SelectTrainersView(CreateView):
+# HACK : should be a UpdateView
+class SelectTeamView(TemplateView):
+    template_name = "battles/battle-team-pokemons.html"
+
+    # TODO : do the validation on the form
+    # TODO : pass error message in context
+
+
+class BattlesView(ListView):  # pylint: disable=too-many-ancestors
```
Comment (rsarai) [inline backend/battles/views.py:43]: This name could be improved to make the behavior more clear, like for instance `BattleListView`

**Review by rsarai — APPROVED** [review]

**Pull request merged**